src/main/java/uc/dboard/submessage/SubmitDBMessageInteractor.java

```diff
@@ -0,0 +1,70 @@
+package uc.dboard.submessage;
+
+import entities.*;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+// Use case layer
+
+public class SubmitDBMessageInteractor implements SubDBMessInputBoundary {
+
+    // no gateway final UserRegisterDsGateway userDsGateway;
+    final SubDBMessPresenter presenter;
+    final MessageFactory messageFactory;
+
+    final SubDBMessDsGateway subDBMessDsGateway;
+
+    final StateTracker stateTracker;
+
+    public SubmitDBMessageInteractor(SubDBMessPresenter presenter,
+                                     MessageFactory messageFactory, SubDBMessDsGateway subDBMessDsGateway, StateTracker stateTracker) {
+        this.presenter = presenter;
+        this.messageFactory = messageFactory;
+        this.subDBMessDsGateway = subDBMessDsGateway;
+        this.stateTracker = stateTracker;
+    }
+
+    @Override
+    public SubDBMessResponseModel submitMessage(SubDBMessRequestModel requestModel) {
+        String solutionId = requestModel.getSolutionId();
+        String testId = subDBMessDsGateway.getDBTestIDFromSolutionID(solutionId);
+        String courseId = subDBMessDsGateway.getDBCourseIDFromTestID(testId);
+        Course course = stateTracker.getCourseIfTracked(courseId);
+
+        //Below Code Traverses through Arraylist to find the test that corresponds to where the message should be added
+
+        //Course should have a tests attribute that is an ArrayList
+        ArrayList<TestDocument> tests = course.tests;
+        TestDocument relatedTest = null;
+        for (TestDocument test:tests){
+            if (test.getId().equals(testId)){
+                relatedTest = test;
```

Comment:
I think the Course entity will have a dictionary mapping the test ID to the test document object. This code can be updated once that's implemented.